src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -175,6 +181,23 @@ private static void addFieldSchema(ResultSetMetaData metadata, int col,
       }
 
       case Types.NUMERIC:
+        if (numericPrecisionMapping) {
+          int precision = metadata.getPrecision(col);
+          if (metadata.getScale(col) == 0 && precision < 20) { // integer
```

Comment:
@clumsy Not sure I understand your response. Isn't the whole point here that you want the `NUMBER` returned for all the `INT` types just work? The mappings you're giving will silently clamp some values that are in the valid range.

I think you might be suggesting that since you're seeing `NUMBER(3,0)` for `TINYINT`, you want to make sure that get's mapped back to `INT8` since that preserves the original width. But I see two problems with that. First, it means that you can lose information when the column actually *is* `NUMBER(3,0)`. Second, I'm pretty sure this isn't even accurate. Some quick googling seems to indicate that in Oracle, the various `INT` types vary between signed and unsigned, so you'd still need to . At least for `TINYINT`, it is actually an unsigned value 0-255, so mapping to `INT8` would actually lose data. It seems like it would be better to map to a type that is guaranteed to capture the full range of values. Lack of signed/unsigned is a tradeoff we made in choosing what to include in the data API based on what different serialization formats support, but it does sometimes mean promoting to a wider type to fully capture the set of possible values.

Unfortunately, I think there are just going to be limits to what we can support if Oracle doesn't export sufficient information to tell between "fake" `NUMBER` columns that are actually `INT` types vs "real" `NUMBER` columns that don't have a constrained set of values.